src/Chainweb/Pact/PactService.hs

```diff
@@ -905,18 +910,19 @@ playOneBlock currHeader plData pdbenv = do
 
     isGenesisBlock = isGenesisBlockHeader currHeader
 
-    go m txs =
-      if isGenesisBlock
+    go m txs = if isGenesisBlock
       then do
         setBlockData currHeader
-        -- reject bad coinbase in genesis
-        execTransactions Nothing m txs (EnforceCoinbaseFailure True) pdbenv
+        -- GENESIS VALIDATE COINBASE: Reject bad coinbase, use date rule for precompilation
+        execTransactions Nothing m txs
+          (EnforceCoinbaseFailure True) (CoinbaseUsePrecompiled False) pdbenv
       else do
         bhDb <- asks _psBlockHeaderDb
         ph <- liftIO $! lookupM bhDb (_blockParent currHeader)
         setBlockData ph
-        -- allow bad coinbase in validate
-        execTransactions (Just bParent) m txs (EnforceCoinbaseFailure False) pdbenv
+        -- VALIDATE COINBASE: back-compat allow failures, use date rule for precompilation
+        execTransactions (Just bParent) m txs
+          (EnforceCoinbaseFailure False) (CoinbaseUsePrecompiled False) pdbenv
```

Comment:
This should always enforce coinbase failure in every case. Precompilation use should defer to date rule (what you have is fine)

Reply:
No -- enforce coinbase failure has ALWAYS needed to be False here otherwise it breaks replay.